server.go

```diff
@@ -99,11 +99,13 @@ type Server struct {
 	shardsBySeriesID map[uint32][]*Shard // shards by series id
 
 	Logger *log.Logger
+
+	authenticationEnabled bool
 }
 
 // NewServer returns a new instance of Server.
-func NewServer() *Server {
-	return &Server{
+func NewServer(authenticationEnabled bool) *Server {
```

Comment:
I made this a function argument specifically because I want the developer to have to make a decision at the point the server is instantiated.  I usually make this distinction when it involves security.